src/validationinterface.cpp

```diff
@@ -150,6 +127,12 @@ void CMainSignals::TransactionAddedToMempool(const CTransactionRef &ptx) {
     });
 }
 
+void CMainSignals::TransactionRemovedFromMempool(const CTransactionRef &ptx) {
```

Comment:
Can you keep the `MemPoolRemovealReason reason` as an argument to this function please? #16688 adds validation interface logging, including logging the reason for the mempool removal.

I also think it makes sense for the client to be notified of the reason for the mempool reason.

Reply:
Can be done in a follow-up, or in #16688 when this gets merged.